compiler/rustc_target/src/target_features.rs

```diff
@@ -273,7 +310,7 @@ const AARCH64_TIED_FEATURES: &[&[&str]] = &[
     &["paca", "pacg"], // Together these represent `pauth` in LLVM
 ];
 
-const X86_ALLOWED_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
+const X86_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
```

Comment:
x87 is noted as a problematic feature both in this file as a comment as well as in the motivation for this PR, but this list does not actually mention x87 (and there isn't another mention outside of the comment.)

Oversight?

Reply:
It was deliberately removed from the PR together with the ARM `fpregs` feature, see [here](https://github.com/rust-lang/rust/pull/129884#issuecomment-2380720755). I plan to further enhance the infrastructure later to support features that are sometimes allowed and sometimes not.

Reply:
Can we leave fixmes for those here?

Reply:
TBH I removed x87 just for consistency with fpregs. The idea is to test the waters with the soft-float features, which we understand best, and then slowly expand -- and see whether there's a big waive of complaints at any point.